.github/workflows/ci.yml

```diff
@@ -33,44 +33,34 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2.0.1
+        with:
+          version: 6.15.1
+
       - name: Set node version to ${{ matrix.node_version }}
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node_version }}
+          cache: "pnpm"
 
-      - name: Get yarn cache directory
-        id: yarn-cache
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - name: Set dependencies cache
-        uses: actions/cache@v2
-        with:
-          path: ${{ steps.yarn-cache.outputs.dir }}
-          key: ${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('yarn.lock') }}
-            ${{ runner.os }}-${{ matrix.node_version }}-
-
-      - name: Versions
-        run: yarn versions
-
-      - name: Install dependencies
-        run: yarn install --frozen-lockfile
+      - name: Install deps
+        run: pnpm install
 
       - name: Build vite
-        run: yarn ci-build-vite
+        run: pnpm run ci-build-vite
```

Comment:
you may actually keep the Yarn syntax here if you wish. `run` is optional with pnpm as well.

Though you still have to escape options with `--` like you did. We will probably make that work without `--` in the future (https://github.com/pnpm/pnpm/issues/3778)

Reply:
Yeah, I'm just doing it for consistency's sake